packages/@react-spectrum/cards/src/GalleryLayout.tsx

```diff
@@ -116,16 +173,35 @@ export class GalleryLayout<T> extends BaseLayout<T> {
     // Compute aspect ratios for all of the items, and the total width if all items were on in a single row.
     let ratios = [];
     let totalWidth = 0;
+    let minRatio = this.minItemSize.width / this.minItemSize.height;
+    let maxRatio = availableWidth / this.minItemSize.height;
+
     for (let node of this.collection) {
       let ratio = node.props.width / node.props.height;
+      if (ratio < minRatio) {
+        ratio = minRatio;
+      } else if (ratio > maxRatio && ratio !== minRatio) {
+        ratio = maxRatio;
+      }
+
+      let itemWidth = ratio * this.minItemSize.height;
       ratios.push(ratio);
-      totalWidth += ratio * this.idealRowHeight;
+      totalWidth += itemWidth;
     }
 
+    totalWidth += this.itemSpacing.width * (this.collection.size - 1);
+
     // Determine how many rows we'll need, and partition the items into rows
     // using the aspect ratios as weights.
-    let rows = Math.max(1, Math.round(totalWidth / availableWidth));
-    let partition = linearPartition(ratios, rows);
+    let rows = Math.max(1, Math.ceil(totalWidth / availableWidth));
+    // if the available width can't hold two items, then every item will get its own row
+    // this leads to a faster run through linear partition and more dependable output for small row widths
+    if (availableWidth <= (this.minItemSize.width * 2) + (this.itemPadding * 2)) {
+      rows = this.collection.size;
+    }
+
+    let weightedRatios = ratios.map(ratio => ratio < this.threshold ? ratio + (0.5 * (1 / ratio)) : ratio);
+    let partition = linearPartition(weightedRatios, rows);
```

Comment:
This partitioning doesn't quite respect the minItemSize.height provided, will need to dig into how the linearPartition works. May require a refactor of these calculations but that can be done after this alpha